### ISSUE TITLE
2021 [NIPS] EditGAN: High-Precision Semantic Image Editing

### ISSUE THREAD

**Issue by pomelyu** [pomelyu/paper-reading-notes#9]
## Introduction

<img width="755" alt="image" src="https://github.com/pomelyu/paper-reading-notes/assets/2679561/e5ac46e8-6792-444c-8f0d-c78167040448">

This paper allows user to edit the images by modifying their segmentation masks. To achieve this, the paper learned the joint distribution between the images and their segmentation masks by training an extra segmentation branch on the pretrained GAN(StyleGAN2). Finally, we can use the optimization approach to find the latent for the modified segmentation mask and the image.

<img width="925" alt="image" src="https://github.com/pomelyu/paper-reading-notes/assets/2679561/cb1765eb-ba62-41b3-b94c-c439a58f1daa">

## Method
- Train the segmentation branch from the pretrained StyleGAN2, so the network generate image $X$ and the corresponding segmentation mask $Y$ from latent $W$
- Find the proper $W + \delta W$ by optimizing the following loss function for the modified segmentation mask $Y_{edit}$ and use $W + \delta W$ to generate the edited image $X_{edit}$
- For the face image, the extra identity constraint from ArcFace is included

<img width="840" alt="image" src="https://github.com/pomelyu/paper-reading-notes/assets/2679561/dbb66e5b-0a53-4d6a-a0f9-3e489ed105b2">

## Highlight
- Only the few(16) labeled samples are needed for training.
- The segmentation map from DatasetGAN[^1] is highly detailed

## Limitation
- **This model consume too many GPU RAM, and can not be ran even on RTX3090 24G**. The segmentation branch predicted the segmentation map from the concatenated feature layers of StyleGAN2 network, and those the input feature map is 6016x512x512, which is not efficient. Actually, they use 10 classifier and ensemble the results as this segmentation branch.
- slow due to the optimization approach. 11.4/18.9s for 30/60 optimization steps(V100). 0.4s if using the editing vector and 4.2/9.5s for 10/30 refinement steps

## Comments

[^1]: Datasetgan: Efficient labeled data factory with minimal human effort.
